server/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshot.java

```diff
@@ -21,132 +21,154 @@
+import org.elasticsearch.snapshots.SnapshotShardFailure;
+import org.elasticsearch.snapshots.SnapshotState;
+import org.jetbrains.annotations.Nullable;
+
+import io.crate.common.collections.Lists;
 import io.crate.metadata.IndexName;
 import io.crate.metadata.IndexParts;
 import io.crate.metadata.PartitionName;
 import io.crate.metadata.RelationName;
 
-public class SysSnapshot {
-
-    private final String uuid;
-    private final String name;
-    private final String repository;
-    private final List<String> concreteIndices;
-    private final List<String> partitionedTables;
-    private final Long started;
-    private final Long finished;
-    private final String version;
-    private final String state;
-    private final String reason;
-    private final int totalShards;
-    private final Boolean includeGlobalState;
-
-    private final List<String> snapshotShardFailures;
-
-    public SysSnapshot(String uuid,
-                       String name,
-                       String repository,
-                       List<String> concreteIndices,
-                       List<String> partitionedTables,
-                       Long started,
-                       Long finished,
-                       String version,
-                       String state,
-                       List<String> snapshotShardFailures,
-                       String reason,
-                       int totalShards,
-                       Boolean includeGlobalState
-                       ) {
-        this.uuid = uuid;
-        this.name = name;
-        this.repository = repository;
-        this.concreteIndices = concreteIndices;
-        this.partitionedTables = partitionedTables;
-        this.started = started;
-        this.finished = finished;
-        this.version = version;
-        this.state = state;
-        this.snapshotShardFailures = snapshotShardFailures;
-        this.reason = reason;
-        this.totalShards = totalShards;
-        this.includeGlobalState = includeGlobalState;
-    }
-
-    public String uuid() {
-        return uuid;
-    }
-
-    public String name() {
-        return name;
-    }
-
-    public String repository() {
-        return repository;
+public record SysSnapshot(String uuid,
+                          String name,
+                          String repository,
+                          List<RelationName> tables,
+                          List<PartitionName> partitions,
+                          List<String> concreteIndices,
+                          Long started,
+                          Long finished,
+                          String version,
+                          String state,
+                          List<String> failures,
+                          String reason,
+                          int totalShards,
+                          Boolean includeGlobalState) {
+
+
+    @Nullable
+    private static IndexMetadata getIndexMetadata(Metadata metadata, String indexName) {
```

Comment:
I'll probably follow up on this later and maybe move it into `Metadata`. There's also a variant in `SnapshotsService` (`resolveIndexByName`) that's related, but I already touch that in https://github.com/crate/crate/pull/18328 - don't want to cause myself conflicts with this PR.